server/src/promotion-manager.js

```diff
@@ -0,0 +1,58 @@
+/* Display Firefox Account onboarding dialog in header below signin button */
+exports.shouldShowOnboardingDialog = function(isOwner, isShotPage) {
+  let show = false;
+  // Exit without showing dialog on non-owner shot page
+  if (isShotPage && !isOwner) {
+    return show;
+  }
+
+  const count = localStorage.hasSeenOnboardingDialog;
+  if (!count) {
+    localStorage.hasSeenOnboardingDialog = 1;
+    show = true;
+  } else if (count < 3) {
+    localStorage.hasSeenOnboardingDialog = parseInt(count, 10) + 1;
+    show = true;
+  }
+  return show;
+};
+
+/* Display edit tool promotion on shot page below edit button */
+exports.shouldShowEditToolPromotion = function(isOwner, enableAnnotations, fxaOnboardingDialog) {
+  // Hide edit tool promotion when showing fxaOnboarding dialog
+  if (!isOwner || !enableAnnotations || fxaOnboardingDialog) {
+    return false;
+  }
+  let show = false;
+  const count = localStorage.hasSeenPromoDialog;
+  if (!count) {
+    localStorage.hasSeenPromoDialog = 1;
+    show = true;
+  } else if (count < 3) {
+    localStorage.hasSeenPromoDialog = parseInt(count, 10) + 1;
+    show = true;
+  }
+  return show;
+};
+
+/* Higlight edit tool icon on shot page */
+exports.shouldHighlightEditIcon = function(isOwner, enableAnnotations) {
+  if (!isOwner) {
+    return false;
+  }
+  const hasSeen = localStorage.hasSeenEditButton;
+  if (!hasSeen && enableAnnotations) {
+    localStorage.hasSeenEditButton = "1";
+  }
+  return !hasSeen;
+};
+
+/* Display signin or upsell ad-banner inside header */
+exports.whichAdBanner = function(shouldGetFirefox, isOwner, hasFxa) {
```

Comment:
Split this into two methods, `shouldShowFirefoxBanner` and `shouldShowFxaBanner`. Call the methods inline instead of comparing the return value. Making this change will give all methods in this class the same boolean return type and keep the function names more uniform.

Reply:
Done